notation/verifier.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notation
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/notaryproject/notation-go"
+	"github.com/notaryproject/notation-go/plugin"
+	notationVerifier "github.com/notaryproject/notation-go/verifier"
+	"github.com/notaryproject/notation-go/verifier/trustpolicy"
+	"github.com/notaryproject/notation-go/verifier/truststore"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/ratify-project/ratify-go"
+	"oras.land/oras-go/v2/registry"
+)
+
+const (
+	notationVerifierType          = "notation"
+	notationSignatureArtifactType = "application/vnd.cncf.notary.signature"
+	maxManifestSizeLimit          = 4 * 1024 * 1024  // 4 MiB
+	maxBlobSizeLimit              = 32 * 1024 * 1024 // 32 MiB
+)
+
+// NewVerifierOptions contains the options for creating a new Notation verifier.
+type NewVerifierOptions struct {
+	// Name is the name of the verifier. Required.
+	Name string
+
+	// TrustPolicyDoc is a trustpolicy.json document. It should follow the spec:
+	// https://github.com/notaryproject/notation-go/blob/release-1.3/verifier/trustpolicy/oci.go#L29
```

Comment:
Never use a branch name in the link. Use a tag or commit hash.

Reply:
Good call, updated to tag instead.